### PR TITLE
disclosure order adressed

### LINKED ISSUE #619: SAP - Rules of Engagement - Disclosures
**Extended Description**
- As an ASAP developer, in order to assist FedRAMP reviewers in ATO review, I want to identify when disclosures have not been provided.

**Acceptance Criteria**
- [ ] All Schematron assertion messages are declarative statements which affirm the positive test outcome.
- [ ] All Schematron assertion diagnostic messages are declarative statements which explain the negative test outcome.
- [ ] The Schematron code has no assertion failures when validated using `src/validations/styleguides/sch.sch` using the `basic` phase.
- [ ] XSpec unit tests for positive and negative Schematron assertion outcomes accompany all Schematron assertions (where feasible).
~~- [ ] terms-and-conditions has a part with an attribute of name with the value of 'disclosures'~~
~~- [ ] There is at least one part with the attribute name with the value of 'disclosure' as a child of the criteria above.~~
- [ ] the disclosure parts have prop elements with numerical sort-id values.  These parts are ordered correctly (document order matches order of sort-id values)

**Definition of Done**
 - [ ] Acceptance criteria met
 - [ ] Unit test coverage of our code > 95%
 - [ ] Automated code quality checks passed
 - [ ] Security reviewed and reported
 - [ ] Reviewed against plain language guidelines
 - [ ] Code must be self-documenting
 - [ ] No local tech debt
 - [ ] Load/performance tests passed – needs to be created/automated
 - [ ] Documentation updated
 - [ ] Architectural Decision Record completed as necessary for significant design choices
 - [ ] PR reviewed & approved
 - [ ] Source code merged

### PR DESCRIPTION
**Pull request by markXLIX**
Closes #619 
disclosure order addressed.

### CODE REVIEW

File: src/validations/rules/sap.sch [hunk 1]
```diff
@@ -478,6 +477,20 @@
                 test="oscal:part[@name eq 'disclosure']">The SAP Rules of Engagement (ROE) Disclosures have one or more detail disclosure
                 statements.</sch:assert>
 
+            <sch:let
+                name="unsorted_disclosures"
+                value="oscal:part[@name eq 'disclosure']/oscal:prop[@name eq 'sort-id']/@value" />
+            <sch:let
+                name="sorted_disclosures"
```
Comment (danielnaab) [inline src/validations/rules/sap.sch:484]: We have a mix between camelCase, dash-case, and snake_case for variables. Might be good to standardize. (not suggesting do that now)

File: src/validations/rules/sap.sch [hunk 2]
```diff
@@ -478,6 +477,20 @@
...
+                value="sort($unsorted_disclosures)" />
+            <sch:assert
+                diagnostics="disclosure-ordered-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAP) §4.17"
+                fedramp:specific="true"
+                id="disclosure-ordered"
+                role="error"
+                test="deep-equal($unsorted_disclosures, $sorted_disclosures)">The SAP terms and conditions disclosures part has disclosure parts that
```
Comment (danielnaab) [inline src/validations/rules/sap.sch:492]:
```suggestion
                test="deep-equal($unsorted_disclosures, $sorted_disclosures)">The SAP terms and conditions disclosures part must have disclosure parts that
```

File: src/validations/test/rules/sap.xspec [hunk 3]
```diff
@@ -1043,6 +1043,77 @@
                         label="that is incorrect" />
                 </x:scenario>
             </x:scenario>
+            <x:scenario
+                label="within a part named 'disclosures'">
+                <x:scenario
+                    label="the parts named 'disclosure' are correctly ordered">
```
Comment (danielnaab) [inline src/validations/test/rules/sap.xspec:1049]:
```suggestion
                    label="when the parts named 'disclosure' are correctly ordered">
```

File: src/validations/test/rules/sap.xspec [hunk 4]
```diff
@@ -1043,6 +1043,77 @@
...
+                                </part>
+                            </terms-and-conditions>
+                        </assessment-plan>
+                    </x:context>
+                    <x:expect-not-assert
+                        id="disclosure-ordered"
+                        label="that is correct" />
+                </x:scenario>
+                <x:scenario
+                    label="the parts named 'disclosure' are not correctly ordered">
```
Comment (danielnaab) [inline src/validations/test/rules/sap.xspec:1083]:
```suggestion
                    label="when the parts named 'disclosure' are not correctly ordered">
```

**Review by danielnaab — CHANGES_REQUESTED** [review]
Looks good, I just had more wording suggestions.

File: src/validations/rules/sap.sch (hunk 1 above)
Comment (markXLIX) [reply src/validations/rules/sap.sch:484]: I agree Dan.  I'll bet most of that is due to 3+ developers on the project.  I know I default to camelCase when I am not keeping dash in mind.

I'll add an issue to READY to check all variables across the schematron near the end of the project.

**Review by danielnaab — APPROVED** [review]

**Pull request merged**